### ISSUE TITLE
DeprecationWarning: on_delete will be a required arg for ForeignKey

### ISSUE THREAD

**Issue by siddharthist** [django-polymorphic/django-polymorphic#297]
Hey there: When testing this with `python3 -Wall`, I get the following warning:
```
/nix/store/5r7yazk3z29584sfrzn80l8bxla953x4-python3.6-django-polymorphic-1.2/lib/python3.6/site-packages/polymorphic/models.py:41: RemovedInDjango20Warning: on_delete will be a required arg for ForeignKey in Django 2.0. Set it to models.CASCADE on models and in existing migrations if you want to maintain the current default behavior. See https://docs.djangoproject.com/en/1.11/ref/models/fields/#django.db.models.ForeignKey.on_delete
  related_name='polymorphic_%(app_label)s.%(class)s_set+')
```
Might be good to future-proof the code that uses `ForeignKey` in this project!

**Comment by hobarrera** [comment]
This seems to be addressed (at least partially) on #298.

**Comment by vdboor** [comment]
That indeed fixes this issue, though I think it should change to `on_delete=models.PROTECT`. Thoughts?

**Comment by meshy** [comment]
I'm not sure if the same logic is applicable, but it looks as though django's [GenericForeignKey proposes `CASCADE`](https://docs.djangoproject.com/en/1.11/ref/contrib/contenttypes/#generic-relations) with its relation to `ContentType`.

**Comment by jleclanche** [comment]
This has been addressed in #298. Changing the trigger is a separate question I guess.
